### ISSUE TITLE
Adding AutoUpdate free (forever?) from GitHub

### ISSUE THREAD

**Issue by Genzzry** [NeverSinkDev/NeverSink-Filter-for-PoE2#37]
If you want to add automatic update for free from GiutHub:

1) Create a text file in your PoE2 filter folder & rename it "update filter.bat" (or "whatever you want it called.bat")
2) Edit the batch (.bat) file and copy/paste the following 2 lines:

```
powershell -Command "Invoke-WebRequest -Uri 'https://raw.githubusercontent.com/NeverSinkDev/NeverSink-PoE2litefilter/0.2.0/NeverSinks Litefilter.filter' -OutFile 'NeverSinks Litefilter.filter'"
pause
```

3) Save the file & exit
4) Thats it... you can now run that file any time you want to update the filter.

* The pause command will let you see that it completed properly and can be removed if you want the output to just go away after being run.

* You can either run that .bat script manually to update the filter file (or put a shortcut to it on your desktop)... or you can have it automatically update every day by running the .bat using Task Scheduler.

100% free auto-update, using GitHub servers instead of Filterblade... for as long as Neversink wants to allow it :)  (for life?)

**Comment by maixmi** [comment]
This just downloads the 0.2.0 version of the filter.

Correct URL for the main
```
powershell -Command "Invoke-WebRequest -Uri 'https://raw.githubusercontent.com/NeverSinkDev/NeverSink-PoE2litefilter/refs/heads/main/NeverSinks Litefilter.filter' -OutFile 'NeverSinks Litefilter.filter'"
pause
```

**Comment by HilltribeG** [comment]
An alternate version I made a few days ago. Displays the version briefly and also launches the game from the default standalone version. Left the steam version in there too. Remove the REM and REM as needed.

```
@echo off
echo Checking directory...
if not exist "%USERPROFILE%\Documents\My Games\Path of Exile 2" (
    echo Creating directory...
    mkdir "%USERPROFILE%\Documents\My Games\Path of Exile 2"
    echo Directory created successfully.
) else (
    echo Directory already exists.
)

echo.
echo Starting download of NeverSink's filter...
cd /d "%USERPROFILE%\Documents\My Games\Path of Exile 2"
curl -L "https://github.com/NeverSinkDev/NeverSink-PoE2litefilter/raw/refs/heads/main/NeverSinks%%20Litefilter.filter" -o "NeverSinks.filter"

if exist "NeverSinks.filter" (
    echo.
    echo Filter downloaded successfully!
    echo Location: %USERPROFILE%\Documents\My Games\Path of Exile 2\NeverSinks.filter
    echo.
    echo Checking filter version...
    findstr "VERSION:" "NeverSinks.filter"
    echo.
    echo Launching Path of Exile 2...
    REM start steam://rungameid/2694490
    start /d "C:\Program Files (x86)\Grinding Gear Games\Path of Exile 2\" PathOfExile.exe
    timeout /t 3
    exit
) else (
    echo.
    echo Error: Filter download failed!
    echo.
    echo Press any key to exit...
    pause >nul
)
```

**Comment by Genzzry** [comment]
> This just downloads the 0.2.0 version of the filter.
> 
> Correct URL for the main
> 
> ```
> powershell -Command "Invoke-WebRequest -Uri 'https://raw.githubusercontent.com/NeverSinkDev/NeverSink-PoE2litefilter/refs/heads/main/NeverSinks Litefilter.filter' -OutFile 'NeverSinks Litefilter.filter'"
> pause
> ```

Thanks for fixing that.
I was in a hurry to get back into game and didn't check the URL.  Just that the command was valid.

Not sure how long this "issue" will stay live though, as Neversink charges people a subscription for auto-update on the Filterblade website.  Having auto-update for free using GitHub servers instead might conflict with the profits from that.

It would be nice to have non-custom / basic versions that stayed free with auto-updates... but you never know.  People have bills to pay, even if they're using free servers :)

**Comment by PhantomGamers** [comment]
> Not sure how long this "issue" will stay live though, as Neversink charges people a subscription for auto-update on the Filterblade website. Having auto-update for free using GitHub servers instead might conflict with the profits from that.

This is nonsense anyway as Neversink has always made the presets available to subscribe to [on the official PoE website](https://www.pathofexile.com/item-filter/ladder/follower) for free auto updates, the subscription is only required for auto updating filters you've customized on the website

**Comment by Genzzry** [comment]
> > Not sure how long this "issue" will stay live though, as Neversink charges people a subscription for auto-update on the Filterblade website. Having auto-update for free using GitHub servers instead might conflict with the profits from that.
> 
> This is nonsense anyway as Neversink has always made the presets available to subscribe to [on the official PoE website](https://www.pathofexile.com/item-filter/ladder/follower) for free auto updates, the subscription is only required for auto updating filters you've customized on the website

Cool, then maybe the auto-update section in the readme will be updated to show that.

Instead of the autoupdate section linking to a video that talks about paying for an autoupdate, it might show this (or another) FREE command that updates the file from this FREE project hosting site?  Of course, keep the other payment options like "buy me a beer", that I and others have used many times... but hopefully "fix" the autoupdate section to highlight that GitHub is a free hosting site, that uses free commands to update from & that subscriptions aren't necessary.

I'm sure someone smarter than me that knows how to use gits push/pull feature can write a short update / guide on how to set up autoupdate via this (or another) command and something like Task Scheduler to enable daily updates... or adding the update command to a batch file so that it updates every time you start PoE2.  This would free up Neversink to work on other things like updating the filter and their website (if they dont have the time / inclination to also "fix" the autoupdate section).

For example, here's my current "PoE2.bat" that auto-updates the loot filter (for free) every time I start PoE2:
```
powershell -Command "Invoke-WebRequest -Uri 'https://raw.githubusercontent.com/NeverSinkDev/NeverSink-PoE2litefilter/refs/heads/main/NeverSinks Litefilter.filter' -OutFile 'NeverSinks Litefilter.filter'"
pause
"C:\Program Files (x86)\Steam\steamapps\common\Path of Exile 2\PathOfExile_x64Steam.exe"
```
The "pause" in there is just so that I can see if there was any errors downloading the file, but can be removed if you want it to just do its thing and don't care about seeing errors, etc.

**Comment by PhantomGamers** [comment]
> Instead of the autoupdate section linking to a video that talks about paying for an autoupdate

The video also mentions the freely available subscription on the poe site lol. But it's an old video and that feature is not yet available on the POE2 website in general

**Comment by goldjee** [comment]
Better use dynamic path to the output folder. This way you can store the script anywhere, not just next to the filter file.

```bat
powershell -Command "Invoke-WebRequest -Uri 'https://raw.githubusercontent.com/NeverSinkDev/NeverSink-PoE2litefilter/refs/heads/main/NeverSinks Litefilter.filter' -OutFile '%userprofile%\Documents\My Games\Path of Exile 2\NeverSinks Litefilter.filter'"
pause
"C:\Program Files (x86)\Steam\steamapps\common\Path of Exile 2\PathOfExile_x64Steam.exe"
```

**Comment by Harlockin** [comment]
> Better use dynamic path to the output folder. This way you can store the script anywhere, not just next to the filter file.
> 
> ```batchfile
> powershell -Command "Invoke-WebRequest -Uri 'https://raw.githubusercontent.com/NeverSinkDev/NeverSink-PoE2litefilter/refs/heads/main/NeverSinks Litefilter.filter' -OutFile '%userprofile%\Documents\My Games\Path of Exile 2\NeverSinks Litefilter.filter'"
> pause
> "C:\Program Files (x86)\Steam\steamapps\common\Path of Exile 2\PathOfExile_x64Steam.exe"
> ```

In powershell you need to use  `$env:UserProfile` instead of %userprofile%

```powershell
powershell -Command "Invoke-WebRequest -Uri 'https://raw.githubusercontent.com/NeverSinkDev/NeverSink-PoE2litefilter/refs/heads/main/NeverSinks Litefilter.filter' -OutFile '$env:USERPROFILE\Documents\My Games\Path of Exile 2\NeverSinks Litefilter.filter'"
pause
 "C:\Program Files (x86)\Steam\steamapps\common\Path of Exile 2\PathOfExile_x64Steam.exe"
```

**Comment by PhantomGamers** [comment]
> In powershell you need to use `$env:UserProfile` instead of %userprofile%

That's a batch script, not a powershell script, thus the `powershell -Command` line at the top. So %userprofile% works fine in that case and your example actually does not.

**Comment by Genzzry** [comment]
I have since added my own customizations that get automatically added into the "current" loot filter using the following:
```
powershell -Command "Invoke-WebRequest -Uri 'https://raw.githubusercontent.com/NeverSinkDev/NeverSink-PoE2litefilter/refs/heads/main/NeverSinks Litefilter.filter' -OutFile 'NeverSinks Litefilter.filter'"
pause
type "Genzzry's Overrides.txt" > modified.filter
type "NeverSinks Litefilter.filter" >> modified.filter
"C:\Program Files (x86)\Steam\steamapps\common\Path of Exile 2\PathOfExile_x64Steam.exe"
```

My overrides are fairly basic and just add a purple border to base types that my Witch might want, but I guess you could use/add any overrides you wanted.  Rares, Uniques, etc all still have their colors, but get a purple border if they're one of the base types listed.

 Here's what "Genzzry's Overrides.txt" looks like:
```
#--------------------------
# Genzzry's Overrides
# (base types found on https://poe2db.tw/us/ )
#--------------------------

# Sceptres
Show 
BaseType "Rattling Sceptre" "Lupine Sceptre" "Ochre Sceptre" "Devouring Sceptre" "Clasped Sceptre" "Devotional Sceptre" "Wrath Sceptre" "Aromatic Sceptre" "Pious Sceptre" "Hallowed Sceptre"
SetBorderColor 255 0 255

# Armor & Energy Shields
Show
BaseType "Sigil Crest Shield" "Emblem Crest Shield" "Jingling Crest Shield" "Sectarian Crest Shield" "Omen Crest Shield" "Wayward Crest Shield" "Seer Crest Shield" "Stoic Crest Shield" "Empyreal Crest Shield" "Deified Crest Shield" "Advanced Blazon Crest Shield" "Advanced Sigil Crest Shield" "Advanced Emblem Crest Shield" "Advanced Jingling Crest Shield" "Advanced Sectarian Crest Shield" "Advanced Omen Crest Shield" "Expert Jingling Crest Shield" "Expert Omen Crest Shield"
SetBorderColor 255 0 255

# Energy Gloves
Show 
BaseType "Torn Gloves" "Sombre Gloves" "Stitched Gloves" "Jewelled Gloves" "Intricate Gloves" "Pauascale Gloves" "Embroidered Gloves" "Adorned Gloves" "Advanced Sombre Gloves" "Advanced Stitched Gloves" "Advanced Jewelled Gloves" "Advanced Intricate Gloves" "Expert Sombre Gloves" "Expert Jewelled Gloves" "Expert Intricate Gloves"
SetBorderColor 255 0 255

# Energy Helms
Show 
BaseType "Twig Circlet" "Wicker Tiara" "Beaded Circlet" "Chain Tiara" "Feathered Tiara" "Gold Circlet" "Coral Circlet" "Jade Tiara" "Noble Circlet" "Twilight Tiara" "Magus Tiara" "Advanced Twig Circlet" "Advanced Wicker Tiara" "Advanced Beaded Circlet" "Advanced Chain Tiara" "Advanced Feathered Tiara" "Expert Wicker Tiara" "Expert Chain Tiara" "Expert Feathered Tiara"
SetBorderColor 255 0 255

# Energy Boots
Show 
BaseType "Straw Sandals" "Wrapped Sandals" "Lattice Sandals" "Silk Slippers" "Feathered Sandals" "Flax Sandals" "Studded Sandals" "Elaborate Sandals" "Advanced Wrapped Sandals" "Advanced Lattice Sandals" "Advanced Silk Slippers" "Advanced Feathered Sandals" "Expert Wrapped Sandals" "Expert Silk Slippers" "Expert Feathered Sandals"
SetBorderColor 255 0 255

# Spirit Amulet
Show 
BaseType == "Solar Amulet"
SetBorderColor 255 0 255

```

This method creates basic auto-updating custom filters from the current one on GitHub

With a little extra "work" (copy & pasting), it could easily be portioned out into seperate files instead of just "Genzzry's Overrides.txt" that could be added depending on your preferences, like "Helm Armor.txt", "Helm Armor+ES.txt", "Shield Armor.txt", "Shield Armor+ES.txt", "Shield Armor+Dodge.txt", etc.  Then it'd only highlight the base types you wanted.

**Comment by goldjee** [comment]
@Genzzry you can combine filters coming from several files. Take a look at `Import` section in the documentation: https://www.pathofexile.com/item-filter/about

Seems there is no need to "physically" merge the files.

**Comment by Genzzry** [comment]
> @Genzzry you can combine filters coming from several files. Take a look at `Import` section in the documentation: https://www.pathofexile.com/item-filter/about
> 
> Seems there is no need to "physically" merge the files.

Would still have to prepend the "import" statements to the github .filter, so it doesn't make much difference if you prepend the "import" line or the actual file... its still one line & has the same effect.

**Comment by NeverSinkDev** [comment]
I want to be as transparent as possible here, so I'd like to elaborate my stance on this:

- Once the PoE ladder filters will be available, a free, economy-updated and auto-updatable version will be available there as well. These were free and will be free forever.

I've been always linking and explaining it's benefits Nothing will change for that in PoE2. 
It's even mentioned first.

![image](https://github.com/user-attachments/assets/7da1ce5f-9412-4ec6-8c48-dda1fc874e5c)

Regarding the FilterBlade auto-update:

- FilterBlade autoupdate has it's own niche: UI edited filters + economy updated data + autoupdate. This is the only paid 'premium' feature we provide and it also causes non-trivial costs for us.
- The donations from Patreon keep the servers running and allow us to spend more time on the projects. 
- It should also be noted that manual updating on FilterBlade has been reduced to a single click and can be done within just 5 seconds nowadays.

Scripts:

When it comes to scripts. I'm not attaching any executable scripts into the downloadable release package. This is a stance I'm not intending to change. I don't think they have any major benefits above the solutions provided above, but at the same time add a lot of potential issues, like user insecurities, machine compatibility etc. I'm much more comfortable with the PoE filter ladder and FilterBlade overall. I will likely add the import command in the future, however it currently causes some issues and this is not a definitive stance.

To sum it up, we currently provide:
- an optimal free solution for people who don't want to edit filters
- a near-instant, free solution for filterblade updating and 100% of other features are also free on filterblade with absolutely minimal ads, despite over 10000 hours of work going into the project.
- a single paid feature with hefty upkeep costs attached on our end

Overall I feel that our implementation is very much in the spirit of how PoE is built and paid for.

I'm leaving this open for the sake of note shutting down the discussion and answering questions, but will likely close it in a couple of days.

**Comment by Genzzry** [comment]
For anyone that was using the .bat file to get free custom updates, the addistion of the single quote in the filenames stops the batch file from woirking as intended, because its a special character.  You'll need to modify the filename to add an extra single-quote like this:
```
powershell -Command "Invoke-WebRequest -Uri 'https://raw.githubusercontent.com/NeverSinkDev/NeverSink-PoE2litefilter/refs/heads/main/NeverSink''s filter 2 - 1-REGULAR.filter' -OutFile 'NeverSink''s filter 2 - 1-REGULAR.filter'"
pause
type "Genzzry's Overrides.txt" > modified.filter
type "NeverSink's filter 2 - 1-REGULAR.filter" >> modified.filter
"C:\Program Files (x86)\Steam\steamapps\common\Path of Exile 2\PathOfExile_x64Steam.exe"

```

Note: The "Genzzry's Overrides.txt" file would still be your own custom modifactions to the loot filter, and what enables it to be a custom auto-updating filter without needing to pay for a subscription.
